src/datasources/targeted-messaging/targeted-messaging.datasource.ts

```diff
@@ -0,0 +1,147 @@
+import { ITargetedMessagingDatasource } from '@/domain/interfaces/targeted-messaging.datasource.interface';
+import { CreateOutreachDto } from '@/domain/targeted-messaging/entities/create-outreach.dto.entity';
+import { CreateTargetedSafesDto } from '@/domain/targeted-messaging/entities/create-targeted-safes.dto.entity';
+import { Outreach } from '@/domain/targeted-messaging/entities/outreach.entity';
+import { Submission } from '@/domain/targeted-messaging/entities/submission.entity';
+import { TargetedSafe } from '@/domain/targeted-messaging/entities/targeted-safe.entity';
+import { ILoggingService, LoggingService } from '@/logging/logging.interface';
+import { asError } from '@/logging/utils';
+import {
+  Inject,
+  NotFoundException,
+  UnprocessableEntityException,
+} from '@nestjs/common';
+import postgres from 'postgres';
+
+export class TargetedMessagingDatasource
+  implements ITargetedMessagingDatasource
+{
+  constructor(
+    @Inject('DB_INSTANCE') private readonly sql: postgres.Sql,
+    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+  ) {}
+
+  async createOutreach(
+    createOutreachDto: CreateOutreachDto,
+  ): Promise<Outreach> {
+    const [outreach] = await this.sql`
+      INSERT INTO outreaches (name, start_date, end_date)
+      VALUES (${createOutreachDto.name}, ${createOutreachDto.startDate}, ${createOutreachDto.endDate})
+      RETURNING *`.catch((err) => {
+      this.loggingService.warn(
+        `Error creating outreach: ${asError(err).message}`,
+      );
+      throw new UnprocessableEntityException('Error creating outreach');
+    });
+
+    return {
+      id: outreach.id,
+      name: outreach.name,
+      startDate: outreach.start_date,
+      endDate: outreach.end_date,
+      created_at: outreach.created_at,
+      updated_at: outreach.updated_at,
```

Comment:
Do you think we should make these camel case for consistency? (I won't mark the others.)

```suggestion
      createdAt: outreach.created_at,
      updatedAt: outreach.updated_at,
```

Reply:
I think it's fine as is.